sklearn/neural_network/extreme_learning_machines.py

```diff
@@ -0,0 +1,629 @@
+
+            if self.verbose:
+                y_scores = self._decision_scores(X[batch_slice])
+
+                if self.batch_size is None:
+                    verbose_string = "Training mean squared error ="
+                else:
+                    verbose_string = "Batch 0, Training mean squared error ="
+
+                print("%s %f" % (verbose_string,
+                                 mean_squared_error(y[batch_slice], y_scores,
+                                                    sample_weight=sw)))
+
+        # Run the least-square algorithm on batch 1, 2, ..., n
+        for batch, batch_slice in enumerate(batches):
+            # Compute hidden activations H_{i} for batch i
+            H_batch = self._compute_hidden_activations(X[batch_slice])
+
+            # Get sample weights (sw) for the batch
+            if sample_weight is None:
+                sw = None
+            else:
+                sw = sample_weight[batch_slice]
+
+            weighted_H_batch = _multiply_weights(H_batch, sw)
+
+            # Update K_{i+1} by H_{i}^T * W * H_{i}
+            self._HT_H_accumulated += safe_sparse_dot(H_batch.T,
+                                                      weighted_H_batch)
+
+            # Update beta_{i+1} by
+            # K_{i+1}^{-1} * H_{i+1}^T * W * (y_{i+1} - H_{i+1} * beta_{i})
+            y_batch = y[batch_slice] - safe_sparse_dot(H_batch,
+                                                       self.coef_output_)
+
+            weighted_y_batch = _multiply_weights(y_batch, sw)
+            Hy_batch = safe_sparse_dot(H_batch.T, weighted_y_batch)
+
+            # Update hidden-to-output coefficients
+            regularized_HT_H = self._HT_H_accumulated.copy()
+            regularized_HT_H.flat[::self.n_hidden + 1] += 1. / self.C
+
+            # It is safe to use linalg.solve (instead of linalg.lstsq
+            # which is slow) since it is highly unlikely that
+            # regularized_HT_H is singular due to the random
+            # projection of the first layer and 'C' regularization being
+            # not dangerously large.
+            self.coef_output_ += linalg.solve(regularized_HT_H, Hy_batch,
+                                              sym_pos=True, overwrite_a=True,
+                                              overwrite_b=True)
+            if self.verbose:
+                y_scores = self._decision_scores(X[batch_slice])
+                print("Batch %d, Training mean squared error = %f" %
+                      (batch + 1, mean_squared_error(y[batch_slice], y_scores,
+                                                     sample_weight=sw)))
+        return self
+
+    def fit(self, X, y, sample_weight=None):
+        """Fit the model to the data X and target y.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape (n_samples, n_features)
+            The input data.
+
+        y : array-like, shape (n_samples,)
```

Comment:
The shape of y is wrong for multi-output regression. Maybe ``fit`` should be in the inheriting classes.